src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -102,6 +104,12 @@ public static class ExpressionMetadataProvider
                 //    m => m.Color.Red (nested property access)
                 //    m => m.Widgets[0].Size (expression ending with property-access)
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
+                if (metadata == null)
+                {
+                    throw new InvalidOperationException(Resources.FormatModelExpression_InvalidProperty(
+                        nameof(ModelMetadata),
```

Comment:
We've done both, I don't have a strong preference if you think one is more correct than the other.